src/components/heartrate/Ppg.cpp

```diff
@@ -52,48 +49,59 @@ int8_t Ppg::Preprocess(float spl) {
 
   auto spl_int = static_cast<int8_t>(spl);
 
-  if (dataIndex < 200)
-    data[dataIndex++] = spl_int;
+  data[dataIndex] = spl_int;
+  dataIndex = (dataIndex + 1) % Ppg::DATA_SIZE;
   return spl_int;
 }
 
-float Ppg::HeartRate() {
-  if (dataIndex < 200)
+int Ppg::HeartRate() {
+  if (data[DATA_SIZE - 1] == 0 || dataIndex % Ppg::UPDATE_HEARTRATE_AFTER != 0) {
     return 0;
+  }
 
   NRF_LOG_INFO("PREPROCESS, offset = %d", offset);
-  auto hr = ProcessHeartRate();
-  dataIndex = 0;
+  int hr = ProcessHeartRate();
   return hr;
 }
-float Ppg::ProcessHeartRate() {
-  auto t0 = Trough(data.data(), dataIndex, 7, 48);
-  if (t0 < 0)
+int Ppg::ProcessHeartRate() {
+  int t0 = Trough(data.data(), DATA_SIZE, 7, 48);
+  if (t0 < 0) {
     return 0;
+  }
 
-  float t1 = t0 * 2;
-  t1 = Trough(data.data(), dataIndex, t1 - 5, t1 + 5);
-  if (t1 < 0)
+  int t1 = t0 * 2;
+  t1 = Trough(data.data(), DATA_SIZE, (t1 - 5), (t1 + 5));
```

Comment:
The added parenthesis aren't necessary here, or the line few lines down.

Reply:
I just added them because clang-tidy was complaining, don't know whats the preferred way.

Reply:
It doesn't complain about that for me.

Reply:
removed in faf2faabea986b1efebb204beab3a4f439008a56